### PR TITLE
(CDAP-5621) Fix the flaky testMapreduceWithDynamicDatasets

### PR DESCRIPTION
**Pull request by chtyim**

### CODE REVIEW

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/batch/MapReduceProgramRunnerTest.java [hunk 1]
```diff
@@ -158,6 +158,10 @@ public void testMapreduceWithDynamicDatasets() throws Exception {
       .build());
     // build runtime args for app
     Map<String, String> runtimeArguments = Maps.newHashMap();
+
+    // Make sure there is only one mapper running at a time since this test has the Mapper writing
+    // to a dataset using increment and the in-memory table doesn't really support concurrent increment
```
Comment (gokulavasan) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/batch/MapReduceProgramRunnerTest.java:163]: Should this be supported? What happens in distributed mode when readlessIncrements are not enabled?

Comment (chtyim) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/batch/MapReduceProgramRunnerTest.java:163]: It should. However, the current dataset implementation structure such that this cannot be supported correctly, due to the `BufferingTable`. It is a known issue and is part of the dataset revamp effort than me and @anew is working on. In the meantime, have this workaround to not having the unit-test fail while still testing what this test was intended for (dynamic dataset access in MR).

**Comment by gokulavasan** [comment]
LGTM 👍

**Comment by anew** [comment]
LGTM

**Comment by anew** [comment]
LGTM

**Comment by chtyim** [comment]
Building http://builds.cask.co/browse/CDAP-DUT4174

**Comment by chtyim** [comment]
Test passed. Squashed and merging.

**Pull request merged**